pandas/tests/groupby/test_raises.py

```diff
@@ -183,6 +180,48 @@ def test_groupby_raises_string(
         ),
     }[groupby_func]
 
+    if using_infer_string:
+        if groupby_func in [
+            "sum",
+            "prod",
+            "mean",
+            "median",
+            "cumsum",
+            "cumprod",
+            "std",
+            "sem",
+            "var",
+            "skew",
+            "quantile",
+        ]:
+            msg = f"str dtype does not support {groupby_func} operations"
+            if groupby_func == "sum":
+                # The object-dtype allows this, StringArray variants do not.
+                klass = TypeError
+            elif groupby_func in ["sem", "std", "skew"]:
+                # The object-dtype raises ValueError when trying to convert to numeric.
+                klass = TypeError
+        elif groupby_func == "pct_change" and df["d"].dtype.storage == "pyarrow":
+            # This doesn't go through EA._groupby_op so the message isn't controlled
+            #  there.
+            import pyarrow as pa
+
+            klass = pa.lib.ArrowNotImplementedError
+            msg = "Function 'divide' has no kernel matching input types"
+        elif groupby_func == "diff" and df["d"].dtype.storage == "pyarrow":
+            # This doesn't go through EA._groupby_op so the message isn't controlled
+            #  there.
+            import pyarrow as pa
+
+            klass = pa.lib.ArrowNotImplementedError
+            msg = "Function 'subtract_checked' has no kernel matching input types"
```

Comment:
Can you add a `# TODO(infer_string)` here? At some point we should avoid bubbling up those pyarrow errors